tools/export/cces/__init__.py

```diff
@@ -0,0 +1,418 @@
+        for flag in flags:
+            if flag.startswith("-gdwarf"):
+                value = prefix + "option.dwarfversion.enumerated.v" + flag[-1]
+                flags.remove(flag)
+                break
+        option = Option("baseId", value)
+        options[prefix + "option.dwarfversion"] = option
+
+    @staticmethod
+    def convert_assembler_options(flags):
+        """
+        Converts the assembler flags into CCES options and removes them
+        from the flags list
+        """
+        options = {}
+
+        # remove these flags without converting to option
+        # since they are added by CCES
+        remove = [ "-x", "assembler-with-cpp" ]
+        for flag in remove:
+            if flag in flags:
+                flags.remove(flag)
+
+        booleans = {"-v": "arm.assembler.option.verbose",
+                    "-g": "arm.assembler.option.debuginfo"}
+
+        for flag in booleans:
+            value = "false"
+            if flag in flags:
+                value = "true"
+                flags.remove(flag)
+            option = Option("baseId", value)
+            options[booleans[flag]] = option
+
+        CCES.convert_common_options("arm.assembler.", options, flags)
+
+        return options
+
+    @staticmethod
+    def convert_compiler_options(flags):
+        """
+        Converts the compiler flags into CCES options and removes them
+        from the flags list
+        """
+        options  = {}
+
+        enable_optimization = "true"
+        value = "arm.base.compiler.option.optimization.og"
+        for flag in flags:
+            if flag.startswith("-O"):
+                value = "arm.base.compiler.option.optimization.o" + flag[2:]
+                if flag[2:] == "0":
+                    enable_optimization = "false"
+                flags.remove(flag)
+                break
+        option = Option("baseId", value)
+        options["arm.base.compiler.option.optimization"] = option
+
+        option = Option("baseId", enable_optimization)
+        options["arm.base.compiler.option.optimization.enable"] = option
+
+        booleans = {"-g": "arm.base.compiler.option.debug",
+                    "-save-temps": "arm.base.compiler.option.savetemps",
+                    "-ffunction-sections": "arm.c.compiler.option.elimination.code",
+                    "-fdata-sections": "arm.c.compiler.option.elimination.data",
+                    "-pedantic": "arm.base.compiler.option.pedantic",
+                    "-pedantic-errors": "arm.base.compiler.option.pedanticerrors",
+                    "-w": "arm.base.compiler.option.inhibitallwarnings",
+                    "-Wall": "arm.base.compiler.option.allwarnings",
+                    "-Wextra": "arm.base.compiler.option.extrawarnings",
+                    "-Werror": "arm.base.compiler.option.warningaserror",
+                    "-Wconversion": "arm.base.compiler.option.conversionwarning"}
+
+        for flag in booleans:
```

Comment:
See the comment about booleans above. Given that this loop appeared twice in this code, could you extract it into it's own function?